src/test/rustdoc-ui/invalid-syntax.stderr

```diff
@@ -127,8 +127,10 @@ LL | /// ```text
 warning: could not parse code block as Rust code
   --> $DIR/invalid-syntax.rs:92:9
    |
-LL | ///     \____/
-   |         ^^^^^^
+LL |   ///     \____/
+   |  _________^
+LL | | ///
+   | |_
```

Comment:
This change looks strange, do you know why it happened?

Reply:
There is a slight difference in the markdown sent to `passes::source_span_for_markdown_range`.

``````
Before this PR: --> 24..30 "```\nlet x = 1;\n```\n\n    \\____/"
With this PR:   --> 24..31 "```\nlet x = 1;\n```\n\n    \\____/\n"
``````

So as you can see, we have an extra backline added now.

Reply:
Explanations about his change: in `impl<'a> FromIterator<&'a DocFragment> for String {`, before we simply collapsed the strings together and added a backline if the "accumulator" wasn't empty. However, we now take into account if a `DocFragment` requires a backline or not which is then added if necessary. This information is stored in the `need_backline` field and computed in the `Attributes::from_ast` method.

So in this case, since this is an indented codeblock, it's a different fragment kind than the one following it, which then adds the backline in the ouput. Here's what the doc fragments looks like:

```
-> DocFragment { line: 0, span: src/test/rustdoc-ui/invalid-syntax.rs:90:1: 90:8 (#0), parent_module: None, doc: " ```", kind: SugaredDoc, need_backline: true, indent: 1 }
-> DocFragment { line: 1, span: src/test/rustdoc-ui/invalid-syntax.rs:91:1: 91:15 (#0), parent_module: None, doc: " let x = 1;", kind: SugaredDoc, need_backline: true, indent: 1 }
-> DocFragment { line: 2, span: src/test/rustdoc-ui/invalid-syntax.rs:92:1: 92:8 (#0), parent_module: None, doc: " ```", kind: SugaredDoc, need_backline: true, indent: 1 }
-> DocFragment { line: 3, span: src/test/rustdoc-ui/invalid-syntax.rs:93:1: 93:4 (#0), parent_module: None, doc: "", kind: SugaredDoc, need_backline: true, indent: 0 }
-> DocFragment { line: 3, span: src/test/rustdoc-ui/invalid-syntax.rs:94:1: 94:15 (#0), parent_module: None, doc: "     \\____/", kind: SugaredDoc, need_backline: true, indent: 1 }
-> DocFragment { line: 4, span: src/test/rustdoc-ui/invalid-syntax.rs:95:1: 95:4 (#0), parent_module: None, doc: "", kind: SugaredDoc, need_backline: false, indent: 0 }
warning: could not parse code block as Rust code
```

Reply:
Wow, that is confusing.

Ok, adding a trailing newline seems like the correct behavior because there's a blank doc-comment after the code block: https://github.com/rust-lang/rust/blob/df2df14424b6fa3b2b2a8a8cdd6cb2307ddbb474/src/test/rustdoc-ui/invalid-syntax.rs#L88-L93. I'm not 100% why the error message changes because of that, but the new error message is reasonable too so I don't think it needs to be debugged.